### DISCUSSION TITLE
Issues with Websocket connection

### DISCUSSION THREAD

**Discussion by eliteTek850** [artisan-roaster-scope/artisan#1755] [Q&A] [answered]
I have built an arduino based roaster controller that communicates via websockets, and would like to use Artisan for data logging.  However, I have having problems getting websockets working properly in Artisan.  Artisan just continually connects, disconnects, connects, disconnects...  In the arduino serial monitor, I can see the connection occurring, and the request being received.  Here is the Artisan ports screen:

![image](https://github.com/user-attachments/assets/3d9fad18-88d0-46f1-b315-9573fb6e5ffb)

And here is the serial monitor showing the connect created, event received, and the output that is sent back:

![image](https://github.com/user-attachments/assets/4803ebab-6d39-4ee6-a83c-03fce160fc81)

And here is the artisan message historys;

![image](https://github.com/user-attachments/assets/4d081970-9685-43b8-a2b8-63e8228f43cd)

Any help would be greatly appreciated.

**Comment by MAKOMO** [comment]
Which version of Artisan are you using?

**Reply by MAKOMO** [reply]
Oh, and what do you see in the ´artisan.log` file (on Windows this is in the following directory: `C:\Users\<USER>\AppData\Local\artisan-scope\Artisan`

Holding ALT & CMD while clicking on the plus icon toggles debug log mode. 

Tick `Logging` to in the Device Assignment dialog to get additional device communication log out put.

**Reply by eliteTek850** [reply, marked as answer]
Looking at the debug log, it looks like the issue may be that the arduino is sending a generic "Hello from RoastPilot Main" message after each connect, and Artisan may not like receiving a message that was not expected.

[artisan.log](https://github.com/user-attachments/files/18011129/artisan.log)

Let me remove that connection message and see if it helps...

**Reply by eliteTek850** [reply]
That was the issue.  I guess when Artisan received a message it didn't expect, it closed the connection.

**Comment by eliteTek850** [comment]
Artisan 3.1.0 (a0c9448) 
Python 3.12.5, Qt 6.7.3, PyQt 6.7.1, Matplotlib 3.9.2, NumPy 2.1.3, SciPy 1.14.1, pymodbus 3.7.4 (1.20.20240911), Yoctopuce 2.0.62875 (2.0.62875)
